mechanicalsoup/browser.py

```diff
@@ -115,3 +117,9 @@ def submit(self, form, url=None, **kwargs):
         response = self.session.send(request)
         Browser.add_soup(response, self.soup_config)
         return response
+
+    def launch_browser(self, soup):
+        """Launch a browser on the page, for debugging purpose."""
+        with tempfile.NamedTemporaryFile(delete=False) as file:
+            file.write(soup.encode())
```

Comment:
It is best to use the output of the HTML parser, because it launches the browser on the page as seen by the parser.

I had to fight bugs where Firefox was seeing something different from html.parser (probably because of a missing closing tag), and some input fields were seen outside a form by html.parser. Launching a browser on soup.encode() showed me directly the input field outside the form, and I could work around it. Launching the browser on the unparsed HTML wouldn't.